website/docs/using-odo/create-component.md

```diff
@@ -3,4 +3,139 @@ title: Create Component
+   ```shell
+   odo create
+   ? Which devfile component type do you wish to create java-springboot
+   ? What do you wish to name the new devfile component java-springboot
+   ? What project do you want the devfile component to be created in myproject
+   Devfile Object Validation
+   ✓  Checking devfile existence [66186ns]
+   ✓  Creating a devfile component from registry: stage [92202ns]
+   Validation
+   ✓  Validating if devfile name is correct [99609ns]
+   ? Do you want to download a starter project (y/N) n
+   ```
+   
+Your Spring Boot component is now ready for use.
+
+## Non-interactively creating a component
+
+To non-interactively create the same component as we created in interactive mode, `cd` into the cloned petclinic project and execute:
+```shell
+odo create java-springboot petclinic -n myproject
+```
+Here `java-springboot` is the type of the component, `petclinic` is the name of the component, and `-n myproject` tells odo to use the project `myproject` for the mentioned `odo create` operation.
+
+## Starter projects
+
+Besides creating a component for an existing code, you could also use "starter project" when creating a component.
+
+Starter projects are example projects developed by the community to showcase the usability of devfiles. An odo user can use these starter projects by running `odo create` command in an empty directory.
+
+### Starer projects in interactive mode
+
+To interactively create a Java Spring Boot component using the starter project, you can follow the below steps:
+```shell
+mkdir myOdoComponent && cd myOdoComponent
+$ odo create
+```
+In the questions that odo asks you next, provide answers like below:
+```shell
+$ odo create
+? Which devfile component type do you wish to create java-springboot
+? What do you wish to name the new devfile component myFirstComponent
+? What project do you want the devfile component to be created in myproject
+Devfile Object Validation
+ ✓  Checking devfile existence [60122ns]
+ ✓  Creating a devfile component from registry: stage [91411ns]
+Validation
+ ✓  Validating if devfile name is correct [35749ns]
+? Do you want to download a starter project Yes
+
+Starter Project
+ ✓  Downloading starter project springbootproject from https://github.com/odo-devfiles/springboot-ex.git [716ms]
+
+Please use `odo push` command to create the component with source deployed
+```
+
+### Starter projects in non-interactive mode
+
+To non-interactively create a Java Spring Boot component using the starter project, you can follow the below steps:
+```shell
+mkdir myOdoComponent && cd myOdoComponent
+odo create java-springboot myFirstComponent --starter
+```
+
+## Push the component to Kubernetes
+
+odo follows a "create & push" workflow for almost all the commands. Meaning, most odo commands won't create resources on Kubernetes cluster unless you run `odo push` command.
```

Comment:
>most odo commands won't create resources on Kubernetes cluster

I am still confused whether we should mention just the K8s cluster or OCP as well.

Reply:
IMO k8s. OCP is for docs.openshift.com, not for upstream.